components/CreateForm.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {useState} from "react";
+import AutoComplete from "react-google-autocomplete"
+import { FormEvent } from "react";
+import { useSession } from "next-auth/react";
+import { Session } from "next-auth";
+
+interface postData {
+    title: string,
+    latitude: number,
+    longitude: number,
+    ownerId: string,
+    description: string,
+    availableSpots: number,
+    email: string | null | undefined,
+    name: string | null | undefined
+    eventTime: string
+}
+
+export interface props {
+    session: Session
+    name: {
+        name: string
+        id: string
+    }
+}
+
+const CreateForm: React.FC<props> = (props) => {
+    const { data: session} = useSession();
+
+    const email = props.session.user?.email
+    const name = props.session.user?.name
+
+    const [isSubmitted, setIsSubmitted] = useState<boolean>(false)
+    const [postData, setPostData] = useState<postData>({
+        title: '',
+        latitude: 0,
+        longitude: 0,
+        ownerId: props.name.id,
+        description: '',
+        availableSpots: 0,
+        email: email,
+        name: name,
+        eventTime: ''
+    })
+
+
+    async function onSubmit(e: FormEvent<HTMLFormElement>) {
+        e.preventDefault()
+        setIsSubmitted(true)
+        const response = await fetch('/api/posts', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(postData)
+        })
+    }
+
+    function handleInput(e: React.ChangeEvent<HTMLInputElement>) {
+        e.preventDefault()
+        const target = e.target as HTMLInputElement
+        const fieldName = target.name;
+        const fieldValue = target.value;
+        
+        if (fieldName === 'availableSpots') {
+            setPostData((prevState) => ({
+                ...prevState,
+                [fieldName] : Number(fieldValue)
+            }))
```

Comment:
It would help to make a helper function to encapsulate ```setPostData((prevState) => ({
                ...prevState,
                [fieldName] : x
            }))``` since you call it multiple times